WordPress/src/main/java/org/wordpress/android/ui/reader/repository/usecases/PostLikeUseCase.kt

```diff
@@ -3,39 +3,55 @@ package org.wordpress.android.ui.reader.repository.usecases
 import kotlinx.coroutines.suspendCancellableCoroutine
 import org.greenrobot.eventbus.Subscribe
 import org.greenrobot.eventbus.ThreadMode.BACKGROUND
+import org.wordpress.android.fluxc.store.AccountStore
 import org.wordpress.android.models.ReaderPost
 import org.wordpress.android.ui.reader.actions.ReaderPostActionsWrapper
+import org.wordpress.android.ui.reader.repository.ReaderRepositoryCommunication
+import org.wordpress.android.ui.reader.repository.ReaderRepositoryCommunication.Error.NetworkUnavailable
+import org.wordpress.android.ui.reader.repository.ReaderRepositoryCommunication.Failure
+import org.wordpress.android.ui.reader.repository.ReaderRepositoryCommunication.SuccessWithData
 import org.wordpress.android.ui.reader.repository.ReaderRepositoryEvent
 import org.wordpress.android.ui.reader.repository.ReaderRepositoryEvent.PostLikeEnded
+import org.wordpress.android.ui.reader.repository.ReaderRepositoryEvent.PostLikeEnded.PostLikeFailure
+import org.wordpress.android.ui.reader.repository.ReaderRepositoryEvent.PostLikeEnded.PostLikeSuccess
+import org.wordpress.android.ui.reader.repository.ReaderRepositoryEvent.PostLikeEnded.PostLikeUnChanged
 import org.wordpress.android.util.EventBusWrapper
+import org.wordpress.android.util.NetworkUtilsWrapper
 import javax.inject.Inject
 import kotlin.coroutines.Continuation
 import kotlin.coroutines.resume
 
-class PostLikeActionUseCase @Inject constructor(
+class PostLikeUseCase @Inject constructor(
     private val eventBusWrapper: EventBusWrapper,
-    private val readerPostActionsWrapper: ReaderPostActionsWrapper
+    private val readerPostActionsWrapper: ReaderPostActionsWrapper,
+    private val accountStore: AccountStore,
+    private val networkUtilsWrapper: NetworkUtilsWrapper
 ) {
-    private val continuations: MutableMap<PostLikeRequest, Continuation<ReaderRepositoryEvent>?> = mutableMapOf()
+    private val continuations:
+            MutableMap<PostLikeRequest, Continuation<ReaderRepositoryCommunication>?> = mutableMapOf()
 
     init {
         eventBusWrapper.register(this)
     }
 
     suspend fun perform(
         post: ReaderPost,
-        isAskingToLike: Boolean,
-        wpComUserId: Long
-    ): ReaderRepositoryEvent {
+        isAskingToLike: Boolean
+    ): ReaderRepositoryCommunication {
+        val wpComUserId = accountStore.account.userId
         val request = PostLikeRequest(post.postId, post.blogId, isAskingToLike, wpComUserId)
 
+        if (!networkUtilsWrapper.isNetworkAvailable()) {
+            return NetworkUnavailable
+        }
+
         if (continuations[request] != null) {
-            return PostLikeEnded.PostLikeUnChanged(
+            return Failure(PostLikeUnChanged(
```

Comment:
AFAIU `continuations[request] != null` means that the request is already in progress right? I feel like it's a bit weird to "fail" in this scenario. Wdyt?

P.S. Btw any reason why we differentiate between Failure and Error?

Reply:
@malinajirka 
> AFAIU `continuations[request] != null` means that the request is already in progress right? I feel like it's a bit weird to "fail" in this scenario. Wdyt?

Yeah, semantics ... the request didn't complete successfully as requested because it was already running, it's a duplicate request that results in unchanged. Is this really a success state? You can argue this both ways - success or error. At the end of the day, it's not an error that needs to be handled by the ui. That's why I used failure to indicate that the request didn't succeed, but I will remove Failure and use success instead.

> Also the structure of ReaderRepositoryEvent and ReaderRepositoryCommunication is a bit confusing. We can easily create invalid states such as Failure(PostLikeSuccess). Wdyt?

`ReaderRepositoryEvent` communicate the responses from the eventBus ... this will get removed when I get rid of failure

Reply:
All updated in commit b89165a. Thanks again

Reply:
>You can argue this both ways - success or error.

Yep, I agree. I decided to go with a new "AlreadyRunning" state in the BlockSiteUseCase. However, it's a minor thing and your solution is imo fine as well. Thanks for the changes.